cpp/arcticdb/async/tasks.hpp

```diff
@@ -396,9 +397,18 @@ struct CopyCompressedInterStoreTask : async::BaseTask {
                 key_segment_pair.set_key(*key_to_write_);
             }
 
-            for (auto & target_store : target_stores_) {
+            for (auto i=0u; i<target_stores_.size(); ++i) {
+                auto& target_store = target_stores_[i];
                 try {
-                    target_store->write_compressed_sync(key_segment_pair);
+                    auto key_segment_pair_copy = key_segment_pair;
+                    if (i < target_stores_.size()-1) {
```

Comment:
Could just iterate with an iterator and check for the end with `std::prev(target_stores_.end())`